src/com/scottyplunkett/server/HTTPRequest.java

```diff
@@ -3,25 +3,52 @@
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Collections;
 
 class HTTPRequest {
     private ArrayList<String> requestContent;
     private String requestLine;
+    private String eTag;
+    private String body;
 
     HTTPRequest(BufferedReader in) throws IOException {
         requestContent = new ArrayList<>();
-        String s;
-        while ((s = in.readLine()) != null) {
-            requestContent.add(s);
-            if (s.isEmpty()) {
+        StringBuilder fullRequest = new StringBuilder();
+        int nextChar;
+        while ((nextChar = in.read()) != -1) {
+            fullRequest.append((char)nextChar);
+            if (!in.ready()) {
```

Comment:
Small thing, but it is not super clear what this line does. I would maybe pull it out into another method so that it is clear.